tools/cli/adminDBCommandsFileWriter.go

```diff
@@ -29,75 +29,92 @@ import (
 )
 
 const (
-	flushThreshold = 100
+	flushThreshold = 50
 )
 
 type (
-	// AdminDBCommandFileWriter is used to batch writes to db scan
-	AdminDBCommandFileWriter interface {
-		AddCorruptedExecution(*CorruptedExecution)
-		AddExecutionCheckFailure(*ExecutionCheckFailure)
+	// AdminDBCorruptedExecutionBufferedWriter is used to buffer writes to file for entities of type CorruptedExecution
+	AdminDBCorruptedExecutionBufferedWriter interface {
```

Comment:
Why do we need two different interfaces and implementations? Does the following one work?
```
BufferedWriter interface {
        Add(interface{})
        Flush()
}
```

Reply:
Good catch :)